cmd/tcl/testworkflow-toolkit/artifacts/junit_post_processor.go

```diff
@@ -0,0 +1,117 @@
+package artifacts
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/fs"
+	"strings"
+	"time"
+
+	"github.com/kubeshop/testkube/pkg/tcl/cloudtcl/data/testworkflow"
+
+	"github.com/pkg/errors"
+
+	"github.com/kubeshop/testkube/cmd/tcl/testworkflow-toolkit/env"
+	cloudexecutor "github.com/kubeshop/testkube/pkg/cloud/data/executor"
+	"github.com/kubeshop/testkube/pkg/filesystem"
+	"github.com/kubeshop/testkube/pkg/ui"
+)
+
+// JUnitPostProcessor is a post-processor that checks XML files for JUnit reports and sends them to the cloud.
+type JUnitPostProcessor struct {
+	fs     filesystem.FileSystem
+	client cloudexecutor.Executor
+}
+
+func NewJUnitPostProcessor(fs filesystem.FileSystem, client cloudexecutor.Executor) *JUnitPostProcessor {
+	return &JUnitPostProcessor{fs: fs, client: client}
+}
+
+func (p *JUnitPostProcessor) Start() error {
+	return nil
+}
+
+// Add checks if the file is a JUnit report and sends it to the cloud.
+func (p *JUnitPostProcessor) Add(path string) error {
+	file, err := p.fs.OpenFileRO(path)
+	if err != nil {
+		return errors.Wrapf(err, "failed to open %s", path)
+	}
+	defer func() { _ = file.Close() }()
+	stat, err := file.Stat()
+	if err != nil {
+		return errors.Wrapf(err, "failed to get file info for %s", path)
+	}
+	if ok := isXMLFile(stat); !ok {
+		return nil
+	}
+	xmlData, err := io.ReadAll(file)
+	if err != nil {
+		return errors.Wrapf(err, "failed to read %s", path)
+	}
+	ok, err := isJUnitReport(xmlData)
+	if err != nil {
+		return errors.Wrapf(err, "failed to check if %s is a JUnit report", path)
+	}
+	if !ok {
+		return nil
+	}
+	fmt.Printf("Processing JUnit report: %s\n", ui.LightCyan(path))
+	if err := p.sendJUnitReport(path, xmlData); err != nil {
+		return errors.Wrapf(err, "failed to send JUnit report %s", stat.Name())
+	}
+	return nil
+}
+
+// sendJUnitReport sends the JUnit report to the Agent gRPC API.
+func (p *JUnitPostProcessor) sendJUnitReport(path string, report []byte) error {
+	ctx, cancel := context.WithTimeout(context.Background(), 15*time.Second)
+	defer cancel()
+	_, err := p.client.Execute(ctx, testworkflow.CmdTestWorkflowExecutionAddReport, &testworkflow.ExecutionsAddReportRequest{
```

Comment:
I thought about it initially, I wanted it to be event-driven and do processing in Cloud, but from what I saw, I think it would require more work and would have more moving pieces, example:
- are the files already uploaded?
- are the files unpacked?
- if no to any of those, we need to persist the logic in db and retry until complete

with current implementation is cheaper to iterate through XML files and check them is any of them a JUnit report, it would be costlier to do the same logic on remote S3, unless we search by regex or something similar, which has edge cases on its own.

Ultimately, I'd like stuff like this to be event-driven and in Cloud, but it would require more work than current. I'd start with this, and gradually move to more event-driven stuff, as also it would unblock graphql work.